Source: written by Pierre Lejeune <superheron@gmail.com>
Upstream: submitted, see issue 7
Reason: fix compilation issues with xerces-c-3 and swig
--- beid-middleware-3.5.2-source-5800.orig/_src/eidmw/applayer/cryptoFramework.cpp	2009-10-28 22:18:33.531272938 +0100
+++ beid-middleware-3.5.2-source-5800/_src/eidmw/applayer/cryptoFramework.cpp	2009-10-28 22:18:49.614022998 +0100
@@ -1764,9 +1764,9 @@
 	XMLByte *pOut=NULL;
 
 	//Encode the baIn
-	pOut=XERCES_CPP_NAMESPACE::Base64::encode((XMLByte *)baIn.GetBytes(),baIn.Size(),&iLenOut);
+	pOut=XERCES_CPP_NAMESPACE::Base64::encode((XMLByte *)baIn.GetBytes(),(size_t)baIn.Size(),(size_t*)&iLenOut);
 
 	//4000000000 is huge, but just checking in case of the unlickely truncation
 	if(!pOut || iLenOut > 4000000000)
 		return false;
 
@@ -1795,7 +1795,7 @@
 	pIn[baIn.Size()]=0;
 
 	//Decode the pIn
-	pOut=XERCES_CPP_NAMESPACE::Base64::decode((XMLByte *)pIn,&iLenOut);
+	pOut=XERCES_CPP_NAMESPACE::Base64::decode((XMLByte *)pIn,(size_t*)&iLenOut);
 	if(!pOut || iLenOut > 4000000000)
 	{
 		delete[] pIn;
--- beid-middleware-3.5.2-source-5800.orig/_src/eidmw/eidlib/eidlib.i	2009-06-17 16:11:08.000000000 +0200
+++ beid-middleware-3.5.2-source-5800/_src/eidmw/eidlib/eidlib.i	2010-01-16 21:07:04.341813800 +0100
@@ -560,7 +560,7 @@
 %typemap(out)          const unsigned char* 
 {
 	$result = jenv->NewByteArray(arg1->Size());
-	jenv->SetByteArrayRegion($result,0,arg1->Size(),(const jbyte*) $1);
+	jenv->SetByteArrayRegion($result,0,arg1->Size(),(jbyte*)$1);
 }
 
 %typemap(javaout)	const unsigned char* 
@@ -1031,7 +1031,7 @@
 		JavaVM *jvm;
 		jenv->GetJavaVM(&jvm);		// recover the JVM pointer
 		
-		SetEventCallback_WrapperCppData* callbackData = new SetEventCallback_WrapperCppData(jvm,(long)arg3);
+		SetEventCallback_WrapperCppData* callbackData = new SetEventCallback_WrapperCppData(jvm,(long)arg3);
 		WrapperCppDataContainer.push_back(callbackData);
 		
 		//------------------------------------------------------		
